src/middleware/guest.js

```
@@ -0,0 +1,33 @@
+// src/middleware/guest.js
+const { verifyAccessToken } = require('../services/tokenService');
+const prisma = require('../services/prisma');
+
+/**
+ * Guest middleware:
+ * - Bloqueia se houver access token válido
+ * - OU se houver refresh cookie válido e não expirado no DB
+ */
+module.exports = async function guest(req, res, next) {
+  try {
+    const bearer = req.header('Authorization')?.replace('Bearer ', '');
+    const cookieAT = req.cookies?.accessToken;
+    const token = bearer || cookieAT;
+
+    if (token) {
+      verifyAccessToken(token);
+      return res.status(403).json({ error: 'Already authenticated' });
+    }
+
+    const rt = req.cookies?.refreshToken;
+    if (rt) {
+      const found = await prisma.refreshToken.findUnique({ where: { token: rt } });
+      if (found && !found.revokedAt && found.expiresAt >= new Date()) {
+        return res.status(403).json({ error: 'Already authenticated' });
+      }
+    }
+
+    return next();
+  } catch {
+    return next(); // se access quebrado, trata como guest
```

The catch block currently swallows any error and treats the request as a guest (`return next()`). That will hide DB errors (Prisma failures) and allow requests through on internal failure. Narrow the error handling: treat JWT verification errors as guest but return 500 on unexpected/internal errors (or at least log and return 500). This prevents accidental bypass on server faults.